flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/operations/PlannerQueryOperation.java

```diff
@@ -43,11 +44,22 @@ public PlannerQueryOperation(RelNode calciteTree) {
 
 		RelDataType rowType = calciteTree.getRowType();
 		String[] fieldNames = rowType.getFieldNames().toArray(new String[0]);
-		TypeInformation[] fieldTypes = rowType.getFieldList()
+		DataType[] fieldTypes = rowType.getFieldList()
```

Comment:
is this change caused by calcite upgrade? does blink planner has similar issue? It's better to add a test to cover this change.

Reply:
I did this change to fix the tests in `LocalExecutorITCase#testViews`, actually i have no idea why the old code base have the correct nullability inference, but based on the current code base, the nullability inference is as expected, and i fix them for `flink-planner`.